dataproc/dataproc_quickstart/quickstart.go

```diff
@@ -0,0 +1,194 @@
+// limitations under the License.
+
+// This sample shows how you can use the Cloud Dataproc Client library to create a
+// Cloud Dataproc cluster
+
+// [START dataproc_quickstart]
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	dataproc "cloud.google.com/go/dataproc/apiv1"
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+	dataprocpb "google.golang.org/genproto/googleapis/cloud/dataproc/v1"
+)
+
+func quickstart(w io.Writer, projectID, region, clusterName, jobFilePath string) error {
+	ctx := context.Background()
+
+	// Create the cluster client
+	endpoint := fmt.Sprintf("%s-dataproc.googleapis.com:443", region)
+	clusterClient, err := dataproc.NewClusterControllerClient(ctx, option.WithEndpoint(endpoint))
+	if err != nil {
+		return fmt.Errorf("Error creating the cluster client: %s", err)
+	}
+
+	// Create the cluster config
+	createReq := &dataprocpb.CreateClusterRequest{
+		ProjectId: projectID,
+		Region:    region,
+		Cluster: &dataprocpb.Cluster{
+			ProjectId:   projectID,
+			ClusterName: clusterName,
+			Config: &dataprocpb.ClusterConfig{
+				MasterConfig: &dataprocpb.InstanceGroupConfig{
+					NumInstances:   1,
+					MachineTypeUri: "n1-standard-1",
+				},
+				WorkerConfig: &dataprocpb.InstanceGroupConfig{
+					NumInstances:   2,
+					MachineTypeUri: "n1-standard-1",
+				},
+			},
+		},
+	}
+
+	// Create the cluster
+	createOp, err := clusterClient.CreateCluster(ctx, createReq)
+	if err != nil {
+		return fmt.Errorf("Error submitting the cluster creation request: %v", err)
+	}
+
+	createResp, err := createOp.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("Error creating the cluster: %v", err)
+	}
+
+	// Output a success message
+	fmt.Fprintf(w, "Cluster created successfully: %q", createResp.ClusterName)
+
+	// Create the job client
+	jobClient, err := dataproc.NewJobControllerClient(ctx, option.WithEndpoint(endpoint))
+
+	// Create the job config
+	submitJobReq := &dataprocpb.SubmitJobRequest{
+		ProjectId: projectID,
+		Region:    region,
+		Job: &dataprocpb.Job{
+			Placement: &dataprocpb.JobPlacement{
+				ClusterName: clusterName,
+			},
+			TypeJob: &dataprocpb.Job_PysparkJob{
+				PysparkJob: &dataprocpb.PySparkJob{
+					MainPythonFileUri: jobFilePath,
+				},
+			},
+		},
+	}
+
+	submitJobResp, err := jobClient.SubmitJob(ctx, submitJobReq)
+	if err != nil {
+		return fmt.Errorf("Error submitting job: %v", err)
+	}
+
+	id := submitJobResp.Reference.JobId
+
+	fmt.Fprintf(w, "Submitted job %q", id)
+
+	// These states all signify that a job has terminated, successfully or not
+	ts := map[dataprocpb.JobStatus_State]bool{
+		dataprocpb.JobStatus_ERROR:     true,
+		dataprocpb.JobStatus_CANCELLED: true,
+		dataprocpb.JobStatus_DONE:      true,
+	}
+
+	// We can create a timeout such that the job gets cancelled if not in a terminal state after a certain amount of time
+	timeout := 5 * time.Minute
+	start := time.Now().UTC()
+
+	state := submitJobResp.Status.State
```

Comment:
Consider checking this inside the for loop and calling `break` when you reach the stopping condition.